src/Adapter/Order/OrderProductQuantityUpdater.php

```diff
@@ -429,4 +455,44 @@ private function getTaxCountry(Cart $cart): Country
 
         return new Country($taxAddress->id_country);
     }
+
+    /**
+     * @param Order $order
+     * @param OrderDetail $orderDetail
+     * @param int $newQuantity
+     *
+     * @throws ProductOutOfStockException
+     */
+    private function assertValidProductQuantity(Order $order, OrderDetail $orderDetail, int $newQuantity)
+    {
+        //check if product is available in stock
+        if (!Product::isAvailableWhenOutOfStock(StockAvailable::outOfStock($orderDetail->product_id))) {
+            $availableQuantity = StockAvailable::getQuantityAvailableByProduct($orderDetail->product_id, $orderDetail->product_attribute_id);
+            $productQuantity = $this->getProductQuantityInOrder($order, (int) $orderDetail->product_id, (int) $orderDetail->product_attribute_id);
```

Comment:
Woups, seems like I forgot to fix this one This was originally done in the handler as well (same code) But it's actually not correct We shouldn't use the quantity in the whole order, but only the one from OrderDetail It's actually weird that this doesn't make tests crash.. This makes me fear that this check is not effective 🤔